crates/sp-domains-fraud-proof/src/lib.rs

```diff
@@ -50,6 +52,18 @@ pub enum FraudProofVerificationInfoRequest {
     DomainRuntimeCode(DomainId),
     /// Domain set_code extrinsic if there is a runtime upgrade at a given consensus block hash.
     DomainSetCodeExtrinsic(DomainId),
+    /// Request to check if particular extrinsic is in range for (domain, bundle) pair at given domain block
+    TxRangeCheck {
```

Comment:
I do not understand. Can you elaborate?

Reply:
The runtime api in domain runtime returns true/false. So, the enum name is `TxRangeCheck`.

Reply:
Here @NingLin-P introduced host function to verify invalid state transition - https://github.com/subspace/subspace/blob/main/crates/sp-domains-fraud-proof/src/host_functions.rs#L38

This and your verification and follow similar request response enums to verify fraud proofs on host that cannot be verified on runtime https://github.com/subspace/subspace/blob/main/crates/sp-domains-fraud-proof/src/host_functions.rs#L35

This makes host function more extensible if we need to add any other fraud proof verifications going forward and does not require a new host function for each verification unit.

Reply:
Are you suggesting that we should be writing two host functions? One to query `TxRange` using consensus client and second host function to check if a particular extrinsic is in range for particular bundle? Wouldn't this be inefficient? 

We had offline discussion where we said to avoid multiple calls to host functions as to<->from communication is costly. And we do not have any use case at the moment for using `TxRange` returned by consensus client at the moment.

Reply:
I do not get you. I am using request response enum only. I cannot utilize @NingLin-P's function as it is since my data requirement is different.

Reply:
Oh. You want the enum to only use data to query and use a separate host function to verify? Hmm. What is the advantage of it? 

Isn't it inefficient to have two separate host functions for this? Where the first host function to query tx range from consensus client. But, this tx range has no use case other than deriving domain tx range. Isn't it better to add a separate host function to only query consensus runtime tx range for domain at later stage?

Reply:
As per our discussion, I am keeping this as it is.